homeassistant/components/notify/smtp.py

```diff
@@ -128,20 +135,26 @@ def send_message(self, message="", **kwargs):
         Build and send a message to a user.
 
         Will send plain text normally, or will build a multipart HTML message
-        with inline image attachments if images config is defined.
+        with inline image attachments if images config is defined, or will
+        build a multipart HTML if html config is defined.
         """
         subject = kwargs.get(ATTR_TITLE, ATTR_TITLE_DEFAULT)
         data = kwargs.get(ATTR_DATA)
 
         if data:
-            msg = _build_multipart_msg(message, images=data.get(ATTR_IMAGES))
+            if ATTR_HTML in data:
+                msg = _build_html_msg(message, data[ATTR_HTML],
+                                      images=data.get(ATTR_IMAGES))
+            else:
+                msg = _build_multipart_msg(message,
+                                           images=data.get(ATTR_IMAGES))
         else:
             msg = _build_text_msg(message)
 
         msg['Subject'] = subject
         msg['To'] = ','.join(self.recipients)
-        msg['From'] = self._sender
-        msg['X-Mailer'] = 'HomeAssistant'
+        msg['From'] = '{} <{}>'.format(self._product_name, self._sender)
+        msg['X-Mailer'] = self._product_name
```

Comment:
The X-Mailer should stick to `HomeAssistant`, as that's the product that emails it.